packages/react-components/react-motion/library/src/hooks/useAnimateAtoms.ts

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import type { AnimationHandle, AtomMotion } from '../types';
+
+/**
+ * In test environments, this hook is used to delay the execution of a callback until the next render. This is necessary
+ * to ensure that the callback is not executed synchronously, which would cause the test to fail.
+ *
+ * @see https://github.com/microsoft/fluentui/issues/31701
+ */
+function useFinishDispatcher() {
+  const [count, setCount] = React.useState(0);
+  const callbackRef = React.useRef<() => void>();
+
+  React.useEffect(() => {
+    if (count > 0) {
+      callbackRef.current?.();
+    }
+  }, [count]);
+
+  return React.useCallback(callback => {
+    callbackRef.current = callback;
+    setCount(v => v + 1);
+  }, []);
+}
+
+function noop() {
+  /* no-op */
+}
+
+/**
+ * @internal
+ */
+export function useAnimateAtoms() {
+  // We don't want to use this hook in any environment other than test.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const dispatchFinishInTest = process.env.NODE_ENV === 'test' ? useFinishDispatcher() : noop;
+
+  return React.useCallback(
+    (
+      element: HTMLElement,
+      value: AtomMotion | AtomMotion[],
+      options: {
+        isReducedMotion: boolean;
+      },
+    ): AnimationHandle => {
+      // Heads up!
+      // Jest uses jsdom as the default environment, which doesn't support the Web Animations API. The same is true for
+      // older browsers that are out of browser support matrix. In these cases, the animation will be a no-op.
+      const SUPPORTS_WEB_ANIMATIONS = typeof element.animate === 'function';
```

Comment:
Could this be moved to module scope and changed to?

`const SUPPORTS_WEB_ANIMATION = typeof document?.body.animate === 'function';`

My understanding is that `animate` is part of the `HTMLElement` interface so you either have it or you don't and that will never change.

A minor thing, but would be nice to avoid repeatedly doing this test if possible.

Reply:
Moved it up 👍